### ISSUE TITLE
icon for fixed drawer not visible when collapsing to tablet and phone sizes

### ISSUE THREAD

**Issue by andhess** [google/material-design-lite#4740]
![image](https://cloud.githubusercontent.com/assets/1474240/18445936/cae95cb4-78d4-11e6-9030-0d274f282436.png)

``` html
<body>
<div id='app'>
<div className="mdl-layout__container">
  <div className="mdl-layout mdl-js-layout mdl-layout--fixed-drawer mdl-layout--fixed-header">
    <header className="mdl-layout__header">
      <div className="mdl-layout__header-row">
        <div className="mdl-layout-spacer"></div>
        <div
          className="mdl-textfield mdl-js-textfield mdl-textfield--expandable
                     mdl-textfield--floating-label mdl-textfield--align-right"
        >
          <label
            className="mdl-button mdl-js-button mdl-button--icon"
            htmlFor="fixed-header-drawer-exp"
          >
            <i className="material-icons">search</i>
          </label>
          <div className="mdl-textfield__expandable-holder">
            <input
              className="mdl-textfield__input"
              type="text"
              name="sample"
              id="fixed-header-drawer-exp"
            />
          </div>
        </div>
      </div>
    </header>
    <div className="mdl-layout__drawer">
      <NavigationLogo companyId={companyId} />
      <nav className="mdl-navigation">
        ...
      </nav>
    </div>
  </div>
</div>
```

I am writing a React application that uses the fixed drawer navigation *after* you navigate past the index route.

After navigating to this index route, I found that I had to manually include a div of `mdl-layout__container` because otherwise the entire application would be tried to be forced into just the header (~50px).

If I directly load a route that uses the navigation there are now 2 consecutive divs of `mdl-layout__container`, which is fine because the side nav button still works. When I start at index and with React navigate the app to a page that uses the fixed drawer, my manual `mdl-layout__container` makes sure the layout is ok, but the generated one is not being added, which I believe is related to `mdl-layout__drawer-button` not being added either.

I'm really stuck here. I've tried a few different workarounds to simulate `mdl-layout__container`, but ultimately, it comes back to the fact that my manual `mdl-layout__container` doesn't seem sufficient, and the `mdl-layout__drawer-button` only appears when the generated `mdl-layout__container` is inserted. I have no idea why `mdl-layout__container` isn't being inserted when I navigate React routes.

**Comment by Garbee** [comment]
The fixed drawer should not be visible on tablet/mobile layouts as it is required by Material Design to shift to the temporary drawer.

Other than that not being visible on those layouts, is there a specific problem you are noticing with the code provided by Material Design Lite? We aren't able to provide integration assistance with third-party code, so all of the React questions are out of scope for this tracker.

Please specify what exactly in the MDL code is giving you a problem with a working reproduction case for example on [codepen](http://codepen.io).

Thank you.

**Comment by andhess** [comment]
Ah, sorry if I was a little unclear above.

When going to tablet/mobile sizes, the icon to toggle the menu is not appearing, thus the drawer cannot be opened. I will try to put something in a codepen to demonstrate

**Comment by dsl101** [comment]
I have what I think might be a related issue. Code which was running last week now is missing the icon, and when expanded on a desktop browser so that the drawer opens, it opens across the title. Previously, the title would show in the header to the right of the drawer.

I think it's related to this: https://github.com/google/material-design-lite/commit/f9b9aadbf4f9441f8b004e9e6a65a8816495a739

If I revert those changes with local CSS, the icon reappears and the title is shown when the drawer expands in a desktop browser.

**Comment by Garbee** [comment]
If someone could please provide actual running code that shows this problem that would be great. There are so many combinations with the layout that trying to scope it down on our end would be very time consuming.

**Comment by dsl101** [comment]
@Garbee Can I PM you a link? It's a live site under development that I'd rather not share publicly. It will take much longer to extract into a fiddle or similar...

**Comment by Garbee** [comment]
Please provide a clean vanilla reproduction cases to ensure third party code isn't conflicting. It should be referenced within the issue so anyone can jump in and help take a look.

Thank you.

**Comment by dsl101** [comment]
OK - will revert once I've had chance to clean it up

**Comment by dsl101** [comment]
Does this help: https://jsfiddle.net/wuspfc9n/3/

That's pretty much what I'm seeing. No icon, and If you extend the width of the output area until the drawer opens, you'll see the title gets covered.

**Comment by Garbee** [comment]
Unable to reproduce it here. Ref: http://codepen.io/Garbee/pen/JRGLdg

First off, you fiddle had broken resource includes. Second, you had 1.2.0 JS with 1.2.1 CSS which isn't right. Always use the same CSS and JS versions. Title is also not covered here.

**Comment by dsl101** [comment]
OK - on our dev site, both CSS and JS were same version (1.2.0), but I couldn't link the CSS from there so I used the customiser to get a CDN link - I hadn't noticed that the version number had changed.

I still do see the problem on our dev site, (using 1.2.1) but I suspect that _is_ conflicts with other stylesheets, or the async loading of stylesheets.

**Comment by Garbee** [comment]
I'm going to go ahead and close this out until a complete reproduction case can be provided that shows where there is a problem with MDL's code. As it is things appear to be working fine for this on our end, which means this falls into support territory which should happen on StackOverflow.

Thank you.

**Comment by dsl101** [comment]
Just in case it helps @andhess, I traced what I think is the cause in my case to asynchronously building the page. If that completes _before_ the pageshow event is triggered, all is well. If however the pageshow event happens before my script has finished constructing the page, all is not well. I will look on SO for advice on dynamically creating content.

**Comment by dsl101** [comment]
And finally, calling `componentHandler.upgradeAllRegistered();` at the end of my code fixed up the icon and title positioning perfectly :). I hadn't seen the section on dynamic sites before (I have only been using MDL for 2 days - that's my excuse).

**Comment by andhess** [comment]
I am using React, and as I dug deeper into this issue found that there are multiple parts that weren't playing well together with MDL. I decided to switch over to the [React-MDL](https://tleunen.github.io/react-mdl/), which is pretty active and supports version 1.2.1. Their support for the Layout component is not ideal - they had to [patch](https://github.com/tleunen/react-mdl/blob/master/extra/layout-patch.diff) MDL to get around the tricky `mdl-layout__container` issue. Since I wanted to use my own custom colors, I actually manually applied their patch changes to my scss files.

I have a solution that works for what I needed. It was a lot more difficult than I expected, so I'm excited to see that the goal of V2 is to include making MDL integrate easier into standard web tools.
